### ISSUE TITLE
monarch.FOCUS.LOST doesn't fire immediately when "Preload" is ticked

### ISSUE THREAD

**Issue by totebo** [britzl/monarch#42]
`monarch.FOCUS.GAINED` fires fine, but not `monarch.FOCUS.LOST`. Is this intentional?

If monarch.FOCUS.LOST should have fired, and the screen is shown again, monarch.FOCUS.GAINED and monarch.FOCUS.LOST are fired in sequence.

**Comment by britzl** [comment]
I'm unable to reproduce this. Can you share an example? Perhaps modify the example included with Monarch?

**Comment by totebo** [comment]
I got a little lost in the example, so I made a minimal project instead. :)

[My Project.zip](https://github.com/britzl/monarch/files/2973217/My.Project.zip)

I realise I use strings instead of hashes, but it doesn't seem to make any difference.

**Comment by britzl** [comment]
Ah, ok, so the problem is that the fg screen gets unloaded before the focus_lost message is received.

**Comment by totebo** [comment]
Is that an issue with my code, or something inside the bowels of Monarch?

**Comment by britzl** [comment]
And since there's no transition on the fg screen it will get immediately unloaded. If there was a transition you'd have enough time to receive the focus_lost message.

**Comment by totebo** [comment]
Is there a different way to solve this, apart from adding a transition?

**Comment by britzl** [comment]
Well, hmm, perhaps monarch should wait a frame before unloading to ensure that the message has had a chance to be processed.

**Comment by totebo** [comment]
🤟

**Comment by totebo** [comment]
I tested this now, and it doesn't appear that the behaviour has changed. Do I need to do something else, aside from "Fetch Libraries"?

**Comment by britzl** [comment]
Oh, ok, let me try again with the project you shared.

**Comment by britzl** [comment]
I added logging and a focus url to both screens:

DEBUG:SCRIPT: SHOWING BACKGROUND
DEBUG:SCRIPT: BG Focus gained, previous screen: 	nil
DEBUG:SCRIPT: SHOWING FOREGROUND
DEBUG:SCRIPT: FG Focus gained, previous screen: 	nil
DEBUG:SCRIPT: BG Focus lost, next screen: 	hash: [fg]
DEBUG:SCRIPT: SHOWING BACKGROUND AGAIN
DEBUG:SCRIPT: FG Focus lost, next screen: 	hash: [bg]
DEBUG:SCRIPT: BG Focus gained, previous screen: 	nil
DEBUG:SCRIPT: SHOWING FOREGROUND AGAIN
DEBUG:SCRIPT: BG Focus lost, next screen: 	hash: [fg]
DEBUG:SCRIPT: FG Focus gained, previous screen: 	nil

As you can see from the log above you get both a focus lost and gained message each time. Modified project:

[totebo-focus.zip](https://github.com/britzl/monarch/files/2983013/totebo-focus.zip)

**Comment by totebo** [comment]
I check the modified project and it works great. Still the same in my game, though, so must be a difference somewhere. I'll investigate.
